cpp/example_code/s3/put_object_acl.cpp

```diff
@@ -0,0 +1,242 @@
+
+    Aws::S3::Model::Owner owner;
+    owner.SetID(ownerID);
+
+    Aws::S3::Model::Grantee grantee;
+    grantee.SetType(SetGranteeType(granteeType));
+
+    if (!granteeEmailAddress.empty()) {
+        grantee.SetEmailAddress(granteeEmailAddress);
+    }
+
+    if (!granteeID.empty()) {
+        grantee.SetID(granteeID);
+    }
+
+    if (!granteeDisplayName.empty()) {
+        grantee.SetDisplayName(granteeDisplayName);
+    }
+
+    if (!granteeURI.empty()) {
+        grantee.SetURI(granteeURI);
+    }
+
+    Aws::S3::Model::Grant grant;
+    grant.SetGrantee(grantee);
+    grant.SetPermission(SetGranteePermission(granteePermission));
+
+    Aws::Vector<Aws::S3::Model::Grant> grants;
+    grants.push_back(grant);
+
+    Aws::S3::Model::AccessControlPolicy acp;
+    acp.SetOwner(owner);
+    acp.SetGrants(grants);
+
+    Aws::S3::Model::PutObjectAclRequest request;
+    request.SetAccessControlPolicy(acp);
+    request.SetBucket(bucketName);
+    request.SetKey(objectKey);
+
+    Aws::S3::Model::PutObjectAclOutcome outcome =
+            s3_client.PutObjectAcl(request);
+
+    if (!outcome.IsSuccess()) {
+        auto error = outcome.GetError();
+        std::cerr << "Error: PutObjectAcl: " << error.GetExceptionName()
+                  << " - " << error.GetMessage() << std::endl;
+    }
+    else {
+        std::cout << "Successfully added an ACL to the object '" << objectKey
+                  << "' in the bucket '" << bucketName << "'." << std::endl;
+    }
+
+    return outcome.IsSuccess();
+}
+
+//! Routine which converts a human-readable string to a built-in type enumeration
+/*!
+ \sa SetGranteePermission()
+ \param access Human readable string.
+*/
+
+Aws::S3::Model::Permission SetGranteePermission(const Aws::String &access) {
+    if (access == "FULL_CONTROL")
+        return Aws::S3::Model::Permission::FULL_CONTROL;
+    if (access == "WRITE")
+        return Aws::S3::Model::Permission::WRITE;
+    if (access == "READ")
+        return Aws::S3::Model::Permission::READ;
+    if (access == "WRITE_ACP")
+        return Aws::S3::Model::Permission::WRITE_ACP;
+    if (access == "READ_ACP")
+        return Aws::S3::Model::Permission::READ_ACP;
+    return Aws::S3::Model::Permission::NOT_SET;
+}
+
+//! Routine which converts a human-readable string to a built-in type enumeration
+/*!
+ \sa SetGranteeType()
+ \param type Human readable string.
+*/
+
+Aws::S3::Model::Type SetGranteeType(const Aws::String &type) {
+    if (type == "Amazon customer by email")
+        return Aws::S3::Model::Type::AmazonCustomerByEmail;
+    if (type == "Canonical user")
+        return Aws::S3::Model::Type::CanonicalUser;
+    if (type == "Group")
+        return Aws::S3::Model::Type::Group;
+    return Aws::S3::Model::Type::NOT_SET;
+}
+
+// snippet-end:[s3.cpp.put_object_acl.code]
+
+/*
+ *
+ * main function
+ *
+ * Prerequisites: An existing bucket that contains the object you want to get
+ * information about the ACL for.
+ *
```

Comment:
* Prerequisites: An existing S3 bucket that contains the object for which to get
 * ACL information.
 *